cloudferrylib/os/actions/check_needed_compute_resources.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+import collections
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils as utl
+
+
+class CheckNeededComputeResources(action.Action):
+    def run(self, **kwargs):
+        info = kwargs['info']
+        objs = info[utl.INSTANCES_TYPE]
+        cnt_map = collections.defaultdict(int)
+        for instance in objs.values():
+            cnt_map[instance['instance']['flavor_id']] += 1
+        needed_cpu = 0
+        needed_ram = 0
+        needed_hdd = 0
+        src_nova = self.src_cloud.resources[utl.COMPUTE_RESOURCE]
+        for id, count in cnt_map.items():
```

Comment:
Here the python built-in function `id` is overridden (https://docs.python.org/2/library/functions.html#id). Please, use another name for this variable.